### PR TITLE
Pan modal coordinators

### PR DESCRIPTION
**Pull request by ajkolean**

### CODE REVIEW

File: Source/PanModal/CoordinatorNavigator+Extension.swift [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+    _ coordinator: BaseCoordinatorPresentable & PanModalCoordinatorNavigator,
+    animated: Bool = true,
+    animationCompletion: AnimationCompletion? = nil
+  ) {
+    addChild(coordinator)
+    presentedCoordinator = coordinator
+    let viewController =
+      coordinator.baseViewController.navigationController
+      ?? coordinator.baseViewController
+    if !(viewController is PanModalPresentable) {
```
Comment (jlaws) [inline Source/PanModal/CoordinatorNavigator+Extension.swift:22]: Can this requirement be part of the PanModalCoordinatorNavigator protocol which would remove the need for the BaseCoordinatorPresentable type requirement above?  Basically have the coordinator protocol this method takes actually have a PanModalPresentable view controller property (probably computed var).  Also, does a UINavigationController automatically conform to PanModalPresentable?

File: Source/PanModal/PanModalCollectionCoordinator.swift [hunk 2]
```diff
@@ -0,0 +1,37 @@
+//
+// Copyright © 2020 Optimize Fitness Inc.
+// Licensed under the MIT license
+// https://github.com/OptimizeFitness/Minerva/blob/master/LICENSE
+//
+
+import Foundation
+import UIKit
+
+public protocol PanModalCoordinatorNavigator {}
```
Comment (jlaws) [inline Source/PanModal/PanModalCollectionCoordinator.swift:10]: Move to a new file and add more type requirements for the view controller type that is required by the PanModal library?

**Review by jlaws — CHANGES_REQUESTED** [review]

File: Source/PanModal/CoordinatorNavigator+Extension.swift (hunk 1 above)
Comment (ajkolean) [reply Source/PanModal/CoordinatorNavigator+Extension.swift]: done

File: Source/PanModal/PanModalCollectionCoordinator.swift (hunk 2 above)
Comment (ajkolean) [reply Source/PanModal/PanModalCollectionCoordinator.swift]: done

**Review by jlaws — APPROVED** [review]

**Pull request merged**